lapce-ui/src/explorer.rs

```diff
@@ -687,52 +689,72 @@ impl Widget<LapceTabData> for FileExplorerFileList {
                     return;
                 }
 
+                match data.config.editor.double_click.as_str() {
+                    "file" => {
+                        double_click_mode = 1;
+                    }
+                    "all" => {
+                        double_click_mode = 2;
+                    }
+                    _ => {
+                        double_click_mode = 0;
+                    }
+                }
+
```

Comment:
this should be enum type rather than int

Reply:
Please see latest commit - is this what you mean?

Reply:
No, `EditorConfig` will be directly deserialized from the file so you can instead automatically generate the conversion using

```rust
#[derive(Serialize, Deserialize)]
enum ClickModes {
    #[serde(rename = "single")]
    SingleClick,
    #[serde(rename = "file")]
    DoubleClickFile,
    #[serde(rename = "all")]
    DoubleClickAll,
}
```

and put the enum directly,

```rust
#[field_names(
        desc = "Use a double click to interact with the file explorer.\nOptions: file, all or Nothing (Default)."
    )]
    pub double_click: ClickModes,
```

and set the default to "single".

This removes the "magic constants" from plaguing the codebase and keep
There doesn't seem to be any error handling yet on the  `ReloadConfig` command (it simply fallback to the default config on failure).

Reply:
Ahhh thanks for the information, as you can probably tell, I'm relatively new to Rust - so this really helps 💪

Ideally a drop-down menu would be even nicer, but unfortunately that was a little too complicated for me at this stage.

Reply:
Yea creating the dropdown menu is quite tricky. Feel free to ping me if you have any more questions.

Reply:
Alright, I have changed it as requested - please take a look.